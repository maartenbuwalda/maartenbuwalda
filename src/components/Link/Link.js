import React from 'react'
import PropTypes from 'prop-types'

const Link = ({ href, children }) => <a href={href} rel="noopener noreferrer" target="_blank">{children}</a>

Link.propTypes = {
  href: PropTypes.string,
  children: PropTypes.node,
}

export default Link
