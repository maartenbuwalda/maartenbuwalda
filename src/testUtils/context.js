const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
