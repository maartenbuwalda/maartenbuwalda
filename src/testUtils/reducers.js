export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
