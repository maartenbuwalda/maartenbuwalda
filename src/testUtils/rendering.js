import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
