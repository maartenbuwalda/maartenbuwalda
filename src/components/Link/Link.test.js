import React from 'react'
import Link from './Link'
import { componentRenders } from '../../testUtils/rendering'

describe('Link component', () => {
  const component = <Link href="test">Test</Link>
  componentRenders(component)
})
