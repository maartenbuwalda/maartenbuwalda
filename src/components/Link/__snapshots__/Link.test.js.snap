// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<a
  href="test"
  rel="noopener noreferrer"
  target="_blank"
>
  Test
</a>
`;
